monitor/statistics.cpp

```diff
@@ -242,9 +242,10 @@ std::string statistics::report(uint64_t categories)
 
 	if (categories & DNET_MONITOR_STATS) {
 #if defined(HAVE_HANDYSTATS) && !defined(HANDYSTATS_DISABLE)
-		rapidjson::Value stats_value(rapidjson::kObjectType);
-		handystats::json::fill(stats_value, allocator, *HANDY_METRICS_DUMP());
-		report.AddMember("stats", stats_value, allocator);
+		rapidjson::Document stats(&allocator);
+		const auto &stats_json = HANDY_JSON_DUMP();
+		stats.Parse<0>(stats_json.c_str());
```

Comment:
There's no need for an intermediate variable, you could just write
```cpp
stats.Parse<0>(HANDY_JSON_DUMP().c_str());
```

Reply:
I use it to lengthen the lifetime of the string returned by `HANDY_JSON_DUMP()` which returns it by value.

Reply:
There's no need to do so as
> All temporary objects are destroyed as the last step in evaluating the full-expression that (lexically) contains the point where they were created

Here full-expression includes `Parse()` call, and the temporary returned from `HANDY_JSON_DUMP()` is valid during the call.

Reply:
fixed